### DISCUSSION TITLE
Is there a way to give placeholder text for the search box?

### DISCUSSION THREAD

**Discussion by vinodsptharsha** [azeezat/react-native-select#34] [Q&A] [answered]
When isSearcheable is enabled, we get a search box in the modal, an input component with borders. End users are getting confused with the box design. A placeholder text for the search box makes it convenient. Is there a way to give placeholder text for the search box?

**Comment by azeezat** [comment, marked as answer]
@vinodsptharsha Great question. A new release has just been published addressing this question
```js
 searchControls={{
              textInputProps: {
                placeholder: 'Search anything here',
                placeholderTextColor: 'gray',
              },
            }} 
```

textInputProps can take the same kind of properties accepted by the base React Native TextInput

**Reply by vinodsptharsha** [reply]
Thanks a lot 👍  I didn't expect a quick release with the requested feature implementation. Now it looks good!!!

**Reply by vinodsptharsha** [reply]
With this update in few places I am getting the below error

<img width="1680" alt="image" src="https://github.com/azeezat/react-native-select/assets/114156301/143a6929-1623-4c51-a641-6ffd7a8c8313">

**Reply by azeezat** [reply]
Thanks for the feedback! Can I see an example of how the component is used? That would be helpful in fixing the issue.

**Reply by vinodsptharsha** [reply]
@azeezat The Issue arises when the data passed to options is not a valid array. Now for an empty array option, I started getting the below error. 

![image](https://github.com/azeezat/react-native-select/assets/114156301/36877afe-d675-476b-83bf-61137e0e1afc)

**Reply by azeezat** [reply]
I think this has been addressed this in the newly published release. Passing in an empty array should show a more reasonable error now.

**Reply by vinodsptharsha** [reply]
@azeezat I think the earlier approach was better where I can pass react query data directly. On initial load, my components will not have any valid array of data. Post mount I will get the proper data from the APIs. Now my entire application broke. To solve this I need to check the array length and then conditionally render the DropDown which now looks ugly. 

A better approach is to allow an empty array or undefined data for options (need to be handled by the package) which may be changed after API calls update.

**Reply by vinodsptharsha** [reply]
Or when invalid data or an empty array is passed to options it is better to show an empty modal with no options to select or an alert message saying no options to select rather than throwing an Error.

**Reply by azeezat** [reply]
@vinodsptharsha Sorry to hear about this. I totally agree with you. This fix should be addressed in the next release

**Comment by vinodsptharsha** [comment]
@azeezat Thank you for restoring the support for the empty array. I have one question! Is there a way, I can put a custom message instead of **no options available** when the array is empty? I have dependent dropdowns, after selecting one option only I will get options for another dropdown. I can show a user message saying please select the first dropdown option first.

**Reply by azeezat** [reply]
You are welcome. I have just published a new release addressing this

```js
 listControls={{
    emptyListMessage: 'No record found',
    ...
 }}
 ```
